### PR TITLE
Fix Vite builder source loader plugin

### PR DESCRIPTION
**Pull request by joshwooding**
Issue: #19554

## What I did

Added getOptions to the mocked class loader.

sourceloader doesn't have any types so I added expect error.

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: code/lib/builder-vite/src/plugins/source-loader-plugin.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ const mockClassLoader = (id: string) => ({
   // eslint-disable-next-line no-console
   emitWarning: (message: string) => console.warn(message),
   resourcePath: id,
+  getOptions: () => ({}),
```
Comment (IanVS) [inline code/lib/builder-vite/src/plugins/source-loader-plugin.ts:14]: It looks like the webpack version has a few options to control things like prettier options, the parser used, etc.  I wonder if there's a way we can get a hold of the `sourceLoaderOptions` from the docs addon, which seems to be the way those are typically set.

Comment (joshwooding) [reply code/lib/builder-vite/src/plugins/source-loader-plugin.ts:14]: @shilman Any ideas here? I don't think the docs preset currently returns this information

**Comment by IanVS** [comment]
Will this still be required if we go with https://github.com/storybookjs/storybook/pull/19680?

**Comment by joshwooding** [comment]
> Will this still be required if we go with https://github.com/storybookjs/storybook/pull/19680?

Nope, this work led to #19680 being done. I guess depending on timeframes of it being merged this can be seen as a "quick fix"

**Comment by shilman** [comment]
Hi Josh, I'm merging #19680 so I'm going to close this. Can you please integrate that instead? Happy to chat about it on Discord and/or pair if it helps.

**Pull request closed without merging**